### PR TITLE
[fix](MySQL) implement `SHOW CHARSET` statement.

### PR DESCRIPTION
**Pull request by liugddx**
## Proposed changes

SHOW CHARSET

Charset|Description  |Default collation |Maxlen|
-------+-------------+------------------+------+
utf8mb4|UTF-8 Unicode|utf8mb4_0900_bin |4     |

https://github.com/dbeaver/dbeaver/pull/22836 needs the pr.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by liugddx** [comment]
run buildall

**Comment by liugddx** [comment]
run buildall

**Comment by liugddx** [comment]
@morrySnow PTAL

**Comment by liugddx** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java [hunk 1]
```diff
@@ -1653,6 +1656,20 @@ private void handleShowAlter() throws AnalysisException {
         resultSet = new ShowResultSet(showStmt.getMetaData(), rows);
     }
 
+    // Show character set.
+    private void handleShowCharset() throws AnalysisException {
+        ShowCharsetStmt showStmt = (ShowCharsetStmt) stmt;
+        List<List<String>> rows = Lists.newArrayList();
+        List<String> row = Lists.newArrayList();
+        // | utf8mb4 | UTF-8 Unicode | utf8mb4_general_ci | 4|
+        row.add("utf8mb4");
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java:1665]: better use value define in `SessionVariable`, such as collationConnection, charsetResults, charsetResults, collationDatabase, etc. You can get it from ctx.getSessionVariables().getxxx.

And please add ut

**Comment by liugddx** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by Yukang-Lian — APPROVED** [review]

**Pull request merged**